File: apps/backend/src/EnterpriseAgentOs.Domain/Agents/IAgentSessionRepository.cs
namespace EnterpriseAgentOs.Domain.Agents;

public interface IAgentSessionRepository
{
    Task<AgentSessionRecord?> GetActiveAsync(Guid agentId, CancellationToken ct = default);
    Task<AgentSessionRecord?> GetByIdAsync(Guid sessionId, CancellationToken ct = default);
    Task<IReadOnlyList<AgentSessionRecord>> ListByAgentAsync(Guid agentId, int limit = 20, CancellationToken ct = default);
    Task<AgentSessionRecord> CreateAsync(AgentSessionRecord session, CancellationToken ct = default);
    Task SaveChangesAsync(CancellationToken ct = default);
    Task<int> CountByAgentAsync(Guid agentId, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentSessionRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentSessionModel;

public interface IAgentSessionRepository {
    CompletableFuture<AgentSessionModel> getActiveAsync(UUID agentId);
    CompletableFuture<AgentSessionModel> getByIdAsync(UUID sessionId);
    CompletableFuture<List<AgentSessionModel>> listByAgentAsync(UUID agentId, int limit);
    CompletableFuture<AgentSessionModel> createAsync(AgentSessionModel session);
    CompletableFuture<Void> saveChangesAsync();
    CompletableFuture<Integer> countByAgentAsync(UUID agentId);
}
